### DISCUSSION TITLE
Associate commitment transaction with a closed channel

### DISCUSSION THREAD

**Discussion by holzeis** [lightningdevkit/rust-lightning#2407] [Ideas / Feature Requests]
At the moment ldk deletes the channel after it has been closed. If I want to keep the channel data around after it has been deleted it's the responsibility of the app layer. But there are a couple of challenges with that approach.

1. The `ChannelClosed` Event does not refer the commitment transaction. So I do know that the channel was closed, but not with what transaction.
2. To get the commitment transaction, I can wait for the `SpendableOutput` Event, but at this point I do not know the Channel this commitment transaction is closing.

I thought about fetching that information through the spent input transaction on the commitment transaction, which has to be the funding transaction and thus get the association with the channel. But this seems to be overly complicated in comparison to ldk simply keeping the channel data after it has been closed and adding the commitment transaction to it once closed. Or at least adding the commitment txid to the `ChannelClosed` Event.

It's very likely that I am missing how to add the association of a commitment transaction to a closed channel more easily. Any advice would be appreciated.

**Comment by wpaulino** [comment]
It should definitely be exposed by LDK somehow. `ChannelClosed` would be a good fit, but unfortunately that event is emitted as soon as we attempt to close the channel. The transaction we attempt to close with may not be the one that ultimately confirms, so ideally we wait until something confirms, maybe even waiting up to `ANTI_REORG_DELAY` confirmations such that we can consider it reorg safe.

**Comment by TheBlueMatt** [comment]
The simplest way is to track the funding outpoint of your channels. With that info, you can always go to a block explorer and ask it what transaction spent the given outpoint. We should, however, provide more info in `SpendableOutput`s, as you tpoint out - https://github.com/lightningdevkit/rust-lightning/issues/2489
